commands/climber/hangingretract.py

```diff
@@ -0,0 +1,71 @@
+from commands2 import CommandBase
+from wpilib import RobotBase
+from subsystems.climbers.leftclimbersubsystem import LeftClimber
+from subsystems.climbers.rightclimbersubsystem import RightClimber
+import constants
+
+# TEMPORARILY OUT OF USE FOR COMP (very sad)
+class HangingLeftClimber(CommandBase):
+    def __init__(self, climber: LeftClimber) -> None:
+        CommandBase.__init__(self)
+        self.setName(__class__.__name__)
+        self.climber = climber
+        self.addRequirements(climber)
+
+    def initialize(self) -> None:
+        self.climber.leftClimb.deactivateBrake()
+
+    def execute(self) -> None:
+        self.climber.leftClimb.setClimberHangingExtension()
+
+    def end(self, _interrupted: bool) -> None:
+        if RobotBase.isReal():
+            self.climber.leftClimb.climbMotor.motor.neutralOutput()
+        else:
+            self.climber.leftClimb.climbMotor.setSpeed(0)
+        self.climber.leftClimb.activateBrake()
+
+    # pylint: disable-next=no-self-use
+    def isFinished(self) -> bool:
+        return (
+            abs(
+                (
+                    self.climber.leftClimb.climbMotor.getPosition()
+                    - constants.kClimbHangingExtension
+                )
+            )
+            < constants.kClimbRetractionMotorThreshold
+        )
+
+
+class HangingRightClimber(CommandBase):
+    def __init__(self, climber: RightClimber) -> None:
+        CommandBase.__init__(self)
+        self.setName(__class__.__name__)
+        self.climber = climber
+        self.addRequirements(climber)
+
+    def initialize(self) -> None:
+        self.climber.rightClimb.deactivateBrake()
+
+    def execute(self) -> None:
+        self.climber.rightClimb.setClimberHangingExtension()
+
+    def end(self, _interrupted: bool) -> None:
+        if RobotBase.isReal():
+            self.climber.rightClimb.climbMotor.motor.neutralOutput()
+        else:
+            self.climber.rightClimb.climbMotor.setSpeed(0)
+        self.climber.rightClimb.activateBrake()
+
+    # pylint: disable-next=no-self-use
+    def isFinished(self) -> bool:
+        return (
+            abs(
+                (
+                    self.climber.rightClimb.climbMotor.getPosition()
+                    - constants.kClimbHangingExtension
+                )
+            )
+            < constants.kClimbRetractionMotorThreshold
+        )
```

Comment:
What is an MR?

Reply:
MR = Merge Request. Technically I should have said PR (pull request) since we're using github instead of gitlab but for our purposes they're basically the same. We won't check in this file in  this PR since it is not necessary for the mid climb.

Reply:
We have some other unused files like complex auto, drivedistance, drivetotarget. If you want we can move it to another branch though

Reply:
Those should be removed as well (in a separate PR not in this branch). For competition, we want the code in main to represent everything we are using and nothing we aren't, so that when (not if) we encounter issues we have the smallest possible set of code to debug.